src/wallet/rpcdump.cpp

```diff
@@ -542,8 +542,11 @@ UniValue dumpprivkey(const JSONRPCRequest& request)
     if (!address.SetString(strAddress))
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Bitcoin address");
     CKeyID keyID;
-    if (!address.GetKeyID(keyID))
+    CScriptID scriptID;
+    if ((!address.GetScriptID(scriptID) || !GetWitnessKeyID(pwalletMain, scriptID, keyID)) &&
+        !address.GetKeyID(keyID)) {
```

Comment:
If I read this right, the logic is the same as previous two cases, except you are preferring witness style fetch over address.GetKeyID fetch. If this is not done for a specific reason, you can
```C++
    if (!FetchWalletKeyID(pwalletMain, address, keyID)) {
```